src/test/c_api_tests.cpp

```diff
@@ -286,6 +288,151 @@ TEST_F(CAPIInference, TensorSetMovedBuffer) {
     ASSERT_EQ(tensor.setBuffer(std::move(buffer)), ovms::StatusCode::DOUBLE_BUFFER_SET);
 }
 
+TEST_F(CAPIInference, NoInputs) {
+    tensor_map_t m;
+    [[maybe_unused]] ovms::ServableMetadata sm("dummy", 1, m, m);
+    OVMS_ServableMetadata* osm = reinterpret_cast<OVMS_ServableMetadata*>(&sm);
+    uint32_t count;
+    ASSERT_EQ(sm.getVersion(), 1);
+    ASSERT_CAPI_STATUS_NULL(OVMS_ServableMetadataGetInputCount(osm, &count));
+    ASSERT_EQ(count, 0);
+}
+
+TEST_F(CAPIInference, InferenceRequest) {
+    InferenceRequest* r = new InferenceRequest("dummy", 1);
+    size_t batchSize;
+    ASSERT_EQ(r->getBatchSize(batchSize, 1), StatusCode::INTERNAL_ERROR);
+    ASSERT_EQ(r->removeInputBuffer("dummy"), StatusCode::NONEXISTENT_TENSOR_FOR_REMOVE_BUFFER);
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_InferenceRequestInputRemoveData(nullptr, "dummy"), StatusCode::NONEXISTENT_PTR);
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_InferenceRequestInputRemoveData(reinterpret_cast<OVMS_InferenceRequest*>(r), nullptr), StatusCode::NONEXISTENT_PTR);
+    delete r;
+}
+TEST_F(CAPIInference, InferenceResponse) {
+    InferenceResponse* r = new InferenceResponse("dummy", 1);
+    int64_t a = 1;
+    EXPECT_THROW(r->addOutput("name", OVMS_DataType::OVMS_DATATYPE_BIN, &a, -1), std::exception);
+    ASSERT_EQ(r->addOutput("n", OVMS_DataType::OVMS_DATATYPE_BIN, &a, 1), StatusCode::OK);
+    InferenceTensor* tensor;
+    const std::string* name;
+    OVMS_InferenceResponse* response = reinterpret_cast<OVMS_InferenceResponse*>(r);
+    const void* voutputData;
+    size_t bytesize = 42;
+    uint32_t outputId = 0;
+    OVMS_DataType datatype = (OVMS_DataType)199;
+    const int64_t* shape{nullptr};
+    size_t dimCount = 42;
+    OVMS_BufferType bufferType = (OVMS_BufferType)199;
+    uint32_t deviceId = 42;
+    const char* outputName = "n";
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_InferenceResponseGetOutput(response, outputId, &outputName, &datatype, &shape, &dimCount, &voutputData, &bytesize, &bufferType, &deviceId), StatusCode::INTERNAL_ERROR);
```

Comment:
Please leave comments next to each invalid call with description why the assert should fail. This will be easier for us in future why the test fails.

Reply:
Why should this test fail with internal error? outputId is correct (0)